src/devices/Mpu9250/Mpu9250.cs

```diff
@@ -0,0 +1,983 @@
+
+                acceBias += accel_temp;
+                gyroBias += gyro_temp;
+            }
+            // Make the average
+            acceBias /= packetCount;
+            gyroBias /= packetCount;
+
+            // bias on Z is cumulative
+            acceBias.Z += acceBias.Z > 0 ? -AccSensitivity : AccSensitivity;
+
+            // Divide by 4 to get 32.9 LSB per deg/s 
+            // Biases are additive, so change sign on calculated average gyro biases
+            rawData[0] = (byte)(((int)(-gyroBias.X / 4) >> 8) & 0xFF);
+            rawData[1] = (byte)((int)(-gyroBias.X / 4) & 0xFF);
+            rawData[2] = (byte)(((int)(-gyroBias.Y / 4) >> 8) & 0xFF);
+            rawData[3] = (byte)((int)(-gyroBias.Y / 4) & 0xFF);
+            rawData[4] = (byte)(((int)(-gyroBias.Z / 4) >> 8) & 0xFF);
+            rawData[5] = (byte)((int)(-gyroBias.Z / 4) & 0xFF);
+
+            // Changes all Gyroscope offsets
+            WriteRegister(Register.XG_OFFSET_H, rawData[0]);
+            WriteRegister(Register.XG_OFFSET_L, rawData[1]);
+            WriteRegister(Register.YG_OFFSET_H, rawData[2]);
+            WriteRegister(Register.YG_OFFSET_L, rawData[3]);
+            WriteRegister(Register.ZG_OFFSET_H, rawData[4]);
+            WriteRegister(Register.ZG_OFFSET_L, rawData[5]);
+
+            // Output scaled gyro biases for display in the main program
+            _gyroscopeBias = gyroBias / GyroSensitivity;
+
+            // Construct the accelerometer biases for push to the hardware accelerometer
+            // bias registers. These registers contain factory trim values which must be
+            // added to the calculated accelerometer biases; on boot up these registers
+            // will hold non-zero values. In addition, bit 0 of the lower byte must be
+            // preserved since it is used for temperature compensation calculations.
+            // Accelerometer bias registers expect bias input as 2048 LSB per g, so that
+            // the accelerometer biases calculated above must be divided by 8.
+
+            // A place to hold the factory accelerometer trim biases
+            Vector3 accel_bias_reg = new Vector3();
+            Span<byte> accData = stackalloc byte[2] { 0, 0 };
+            // Read factory accelerometer trim values
+            ReadByteArray(Register.XA_OFFSET_H, accData);
+            accel_bias_reg.X = BinaryPrimitives.ReadUInt16BigEndian(accData);
+            ReadByteArray(Register.YA_OFFSET_H, accData);
+            accel_bias_reg.Y = BinaryPrimitives.ReadUInt16BigEndian(accData);
+            ReadByteArray(Register.ZA_OFFSET_H, accData);
+            accel_bias_reg.Z = BinaryPrimitives.ReadUInt16BigEndian(accData);
+
+            // Define mask for temperature compensation bit 0 of lower byte of
+            // accelerometer bias registers
+
+            uint mask = 0x01;
+            // Define array to hold mask bit for each accelerometer bias axis
+            byte[] mask_bit = new byte[3];
```

Comment:
NIT: let's avoid allocations in here as well and use Span<byte> with stackalloc.